lib/std/http/method.zig

```diff
@@ -0,0 +1,65 @@
+//! HTTP Methods
+//! https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+
+// Style guide is violated here so that @tagName can be used effectively
+/// https://datatracker.ietf.org/doc/html/rfc7231#section-4 Initial definiton
+/// https://datatracker.ietf.org/doc/html/rfc5789#section-2 PATCH
+pub const Method = enum {
+    GET,
+    HEAD,
+    POST,
+    PUT,
+    DELETE,
+    CONNECT,
+    OPTIONS,
+    TRACE,
+    PATCH,
+
+    /// Returns true if a request of this method is allowed to have a body
+    /// Actual behavior from servers may vary and should still be checked
+    pub fn requestHasBody(self: Method) bool {
+        return switch (self) {
+            .POST, .PUT, .PATCH => true,
+            .GET, .HEAD, .DELETE, .CONNECT, .OPTIONS, .TRACE => false,
+        };
+    }
+
+    /// Returns true if a response to this method is allowed to have a body
+    /// Actual behavior from clients may vary and should still be checked
+    pub fn responseHasBody(self: Method) bool {
+        return switch (self) {
+            .GET, .POST, .DELETE, .CONNECT, .OPTIONS, .PATCH => true,
+            .HEAD, .PUT, .TRACE => false,
```

Comment:
I'm not sure why `PUT` is in this list?

Infact *most* `PUT` responses I've seen have a body on them.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT

Reply:
Here's an [example of PUT returning a body.](https://github.com/tick/tick-api/blob/master/sections/entries.md#update-entry)

I would still say the stdlib should follow the http standard, but it should not block the ability to get a response body from a PUT request.

Reply:
it wouldnt stop a client from receiving a PUT body, but it could raise a warning in the server implementation from sending a body